docs/python/Runner.md

```diff
@@ -0,0 +1,85 @@
+
+Every module, for every objective it supports, defines a class implementing abstract class `ITest`.
+
+## Supported Objective Types
+Currently supported objective types:
+
+| Full Name | Short Name |
+| -- | -- |
+| Gaussian Mixture Model Fitting | GMM |
+| Bundle Adjustment| BA |
+| Hand Tracking | Hand |
+| Long short-term memory | LSTM |
+
+
+## Command Line
+
+```powershell
+python3 /path/to/repo/src/python/runner/main.py test_type module_path input_filepath output_dir minimum_measurable_time nruns_F nruns_J time_limit [-rep]
+```
+
+ - `test_type` - The short type name of the objective function. It may also be equal to "Hand-Complicated" that designates the complicated case of the "Hand" objective, where the variable U is considered (see [srajer-autodiff-screen.pdf](../../Documents/srajer-autodiff-screen.pdf), page 5).
+ - `module_path` - an absolute or relative path to a `.py` file containing the module to be benchmarked.
+ - `input_filepath` - an absolute or relative path to the input file.
+ - `output_dir` - a directory where the output files should be stored.
+ - `minimum_measurable_time` - minimum time that the computation needs to run to produce a reliable measurement.
+ - `nruns_F` - maximum number of times to run the computation of the objective function for timing.
+ - `nruns_J` - maximum number of times to run the computation of the considered derivative (gradient or Jacobian) for timing.
+ - `time_limit` - soft (see [Methodology.md](../Methodology.md)) time limit for benchmarking the computation of either the objective function or its gradient/Jacobian.
+ - `-rep` *(applicable only for GMM)* - if enabled, all input data points are expected to have one shared value.
+
+## Adding new modules
+(see [Modules.md](./Modules.md))
+
+## Adding new objective types
+
+1. Define `TInput` and `TOutput` classes in the `/src/python/shared/TData.py` file where `T` is the name of a new objective. 
+
+    `TInput` is the data type for the new objective's inputs, `TOutput` is a structure that contains outputs of both new objective function and its target jacobian. Also `TOutput` contains `save_output_to_file` function
+
+    ```python
+    def save_output_to_file(
+        self,
+        output_prefix,
+        input_basename,
+        module_basename
+    )
+    ```
+
+    Which saves results of computations stored in a structure of the TOutput type to the output files:
+
+    - `output_prefix + input_basename + "_F_" + module_basename + ".txt"` - stores the value of the objective function
+    - `output_prefix + input_basename + "_J_" + module_basename + ".txt"` - stores the value of the objective function derivative
+
+      The format of the output files is specific for each objective type.
+
+1. In the `input_utils` (`/src/python/shared/input_utils.py`) add the following function:
+    ```python
+    def read_T_instance(fn):
+    ```
+    It open input file `fn` and loads data to the structure of the `TInput` type.
```

Comment:
Misprint: `It open` instead of `It opens`

Reply:
fixed